### PR TITLE
Disable non index pattern datasources redirection when new nav is on and query enhancement is off

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

Disable non index pattern datasources redirection when new nav is on and query enhancement is off.

### Issues Resolved

N/A

## Changelog
- fix: Disable non index pattern datasources redirection when new nav is on and query enhancement is off

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx [hunk 1]
```diff
@@ -136,12 +139,14 @@ export const DataSourceSelectable = ({
         setDataSourceOptionList(
           consolidateDataSourceGroups(
             dataSetResults as DataSetWithDataSource[],
-            dataSources.filter((ds) => !ds.getMetadata().ui.selector.displayDatasetsAsSource)
+            newHomePageEnabled && !queryEnhancementsEnabled
+              ? []
+              : dataSources.filter((ds) => !ds.getMetadata().ui.selector.displayDatasetsAsSource)
```
Comment (joshuali925) [inline src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx:144]: what does this do? title says s3 but logic doesn't seem s3 specific

**Review by joshuali925 — COMMENTED** [review]
are there conflicts in s3 redirection when new features are enabled? and does it need tests to ensure the behavior?

File: src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx (hunk 1 above)
Comment (mengweieric) [reply src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx:144]: I should change the title to make it more clear.

**Review by ashwin-pc — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Comment by mengweieric** [comment]
unable to verify, please reopen when sees the issue.

**Comment by mengweieric** [comment]
Put the notes here: This PR was raised and determined, after spoke to @ashwin-pc , as an Ad hoc, workaround fix at the time for redirection on flint datasource to Log explorer with the conditions: New navigation is on and query enhancement is off. The PR was the fix attempt to be included for 2.16 but was not able to make it. The issue happened due to the new nav changes broke the previously working redirection. Closed this PR as we should still consider fix the redirection rather than filter out non-index patterns datasources as long as flint usage is still a valid use case when that conditions are active. Please reopen it if we still want this as the long term changes.

**Pull request closed without merging**